code/src/reducers/game.js

```diff
@@ -0,0 +1,62 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+export const game = createSlice({
+  name: 'game',
+  initialState: {
+    username: '',
+    currentGameState: {},
+    coordinatesArray: [],
+    pastActions: [],
+    positionHistory: [],
+    squares: ['0,3', '1,3', '0,2', '1,2', '0,1', '1,1', '0,0', '1,0']
+  },
+  reducers: {
+    addUserName: (store, action) => {
+      const { username } = action.payload;
```

Comment:
No the username is set as an object, but is specified as a string in the intitialState. 
This could be written as : `state.username = action.payload;` instead.